### PR TITLE
Stp/remove trees

### PR DESCRIPTION
**Pull request by Yuhanun**
Trees have been completely removed from codebase, no need to merge yet, just did this because i wanted to know how many files they were. 

(315 -> 204 ish translation units)

Started resolving code analysis issues in clion aswel but too much for 1 PR.

### CODE REVIEW

File: src/utilities/RobotDealer.cpp [hunk 1]
```diff
@@ -3,7 +3,6 @@
 #include "coach/BallplacementCoach.h"
 #include "coach/PassCoach.h"
 #include "control/ControlUtils.h"
-#include "treeinterp/BTFactory.h"
 
 namespace rtt::ai::robotDealer {
```
Comment (RobotJente) [inline src/utilities/RobotDealer.cpp]: I think this should also be removed

**Comment by RobotJente** [comment]
Waiting for #835 to be merged

File: src/utilities/RobotDealer.cpp (hunk 1 above)
Comment (Yuhanun) [reply src/utilities/RobotDealer.cpp]: robotdealer? alright

**Comment by timovdk** [comment]
You can also remove all the controllers except the files in control/positionControl, the control/ControlUtils.h and control/RobotCommand.h files.

You can remove all coaches (the complete folder)

You can remove world/BallPossession.h/cpp 

You can remove world_new/RobotControllers.h/cpp

You also have some unused .cpp files still there, like RobotDealer.cpp

**Comment by timovdk** [comment]
> You can also remove all the controllers except the files in control/positionControl, the control/ControlUtils.h and control/RobotCommand.h files.
> 
> You can remove all coaches (the complete folder)
> 
> You can remove world/BallPossession.h/cpp
> 
> You can remove world_new/RobotControllers.h/cpp
> 
> You also have some unused .cpp files still there, like RobotDealer.cpp

Already did this!

**Comment by Yuhanun** [comment]
Been busy with exams, sorry @timovdk

**Comment by timovdk** [comment]
> Been busy with exams, sorry @timovdk 

Don't worry about it, heard it from Jesse so I removed the things I listed. Good luck with your exams!

**Review by RobotJente — APPROVED** [review]

**Pull request merged**